### ISSUE TITLE
Issues in spatial analysis with spaceranger2.0.0 outputs, and potential solutions

### ISSUE THREAD

**Issue by linhuawang** [scverse/scanpy#2345]
#### Problem
File `tissue_positions_list.csv` not found when running `sc.read_visium`
This issue was caused by the file name change by `spaceranger`. 

#### Solution
Make a copy of the `outs/spatial/tissue_positions.csv` and name it as `outs/spatial/tissue_positions_list.csv`

#### Problem
The following error messages appeared when running `sc.pl.spatial`:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [4], line 1
----> 1 sc.pl.spatial(adata, color=adata.var_names.tolist()[0], img_key='hires', scale_factor=None)

File scanpy/plotting/_tools/scatterplots.py:1003, in spatial(adata, basis, img, img_key, library_id, crop_coord, alpha_img, bw, size, scale_factor, spot_size, na_color, show, return_fig, save, **kwargs)
   1000     cmap_img = None
   1001 circle_radius = size * scale_factor * spot_size * 0.5
-> 1003 axs = embedding(
   1004     adata,
   1005     basis=basis,
   1006     scale_factor=scale_factor,
   1007     size=circle_radius,
   1008     na_color=na_color,
   1009     show=False,
   1010     save=False,
   1011     **kwargs,
   1012 )
   1013 if not isinstance(axs, list):
   1014     axs = [axs]

File scanpy/plotting/_tools/scatterplots.py:392, in embedding(adata, basis, color, gene_symbols, use_raw, sort_order, edges, edges_width, edges_color, neighbors_key, arrows, arrows_kwds, groups, components, dimensions, layer, projection, scale_factor, color_map, cmap, palette, na_color, na_in_legend, size, frameon, legend_fontsize, legend_fontweight, legend_loc, legend_fontoutline, colorbar_loc, vmax, vmin, vcenter, norm, add_outline, outline_width, outline_color, ncols, hspace, wspace, title, show, save, ax, return_fig, **kwargs)
    389         # if user did not set alpha, set alpha to 0.7
    390         kwargs['alpha'] = 0.7 if alpha is None else alpha
--> 392     cax = scatter(
    393         coords[:, 0],
    394         coords[:, 1],
    395         marker=".",
    396         c=color_vector,
    397         rasterized=settings._vector_friendly,
    398         norm=normalize,
    399         **kwargs,
    400     )
    402 # remove y and x ticks
    403 ax.set_yticks([])

File scanpy/plotting/_utils.py:1111, in circles(x, y, s, ax, marker, c, vmin, vmax, scale_factor, **kwargs)
   1105 # You can set `facecolor` with an array for each patch,
   1106 # while you can only set `facecolors` with a value for all.
   1107 if scale_factor != 1.0:
-> 1108     x = x * scale_factor
   1109     y = y * scale_factor
   1110 zipped = np.broadcast(x, y, s)

TypeError: can't multiply sequence by non-int of type 'float'
```

#### Solution

This issue was caused by non-integer values in `adata.obsm['spatial']` after reading using `sc.read_visium`. Either of the following two solutions works:

1.  `adata.obsm['spatial'] = adata.obsm['spatial'].astype(int)`
2.  Add the following line 1108 - 1110 in `scanpy/plotting/_utils.py`:

```python
1107     if scale_factor != 1.0:
1108         if not isinstance(x[0], int) or not isinstance(y[0], int):
1109             x = x.astype(int)
1110             y = y.astype(int)
1111         x = x * scale_factor
1112         y = y * scale_factor
```

Personally, I suggest the second solution.

Also, there is a post talking about not maintaining the spatial module in `scanpy`. 

#### Versions
* `scanpy==1.9.1`
* `spaceranger==2.0.0`

**Comment by johnyaku** [comment]
We're experiencing the same problem @linhuawang 

We can hack our way around it, but it would be neater if `scanpy` dealt with the new files out of the box.

You seem to be fully across the problem -- could you make a PR @linhuawang ?

**Comment by johnyaku** [comment]
Don't worry @linhuawang, I've made a PR myself #2458

**Comment by giovp** [comment]
Hi @johnyaku and @linhuawang ,
apologies for late reply, we just merged #2424 in main already. Should be fixed! Will close but feel free to reopen if not fixed
